src/fides/api/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -75,6 +83,44 @@ def get_connector_templates(self) -> Dict[str, ConnectorTemplate]:
         return self.templates
 
 
+class CustomConnectorTemplateLoader(ConnectorTemplateLoader):
+    """
+    Loads custom connector templates defined in the custom_connector_template database table.
+    """
+
+    def __init__(self) -> None:
+        logger.info("Loading connectors templates from the database")
+        self.templates: Dict[str, ConnectorTemplate] = {}
+        for template in CustomConnectorTemplate.all(db=get_api_session()):
+            try:
```

Comment:
the checks/error handling around template instantiation/registration that are done here are not being invoked in the dynamic registration. i know the already instantiated `ConnectorTemplate` is being passed into the dynamic registration method, so it may not belong in that method specifically, but i do think we need to abstract out this instantiation/initialization logic here away from the `CustomConnectorTemplateLoader` constructor to ensure that we are running through the same checks whether the template is loaded via a persisted db record or via the dynamic registration.